src/web/lib/useHasBeenSeen.ts

```diff
@@ -11,11 +12,16 @@ const useHasBeenSeen = (options: IntersectionObserverInit) => {
             observer.current.disconnect();
         }
 
-        observer.current = new window.IntersectionObserver(([entry]) => {
-            if (entry.isIntersecting) {
-                setHasBeenSeen(true);
-            }
-        }, options);
+        // Debounce to ensure the node is seen for at least 200ms before the
+        // boolean is set to true
```

Comment:
:+1: I think this comment is justified and useful here. I think it also highlights the difference between what we're doing here vs what we're doing on frontend. Here the element has to intersect for at least 200ms before we fire the callback (you could still be scrolling, the callback will still fire is my understanding) whereas on frontend I think you have to have _stopped_ scrolling for 200ms and the element be intersecting for the callback to fire.

Reply:
That's my understanding too. This fix should take us some way there but not completely. For parity with Frontend, we'd need to actually replicate their behaviour and implement something new using scroll callbacks. But this approach is so much better it's worth trying a couple of fixes first to see how impactful they are.